### ISSUE TITLE
Sometimes a HTML page is served back ; Specify user agent?

### ISSUE THREAD

**Issue by yassinphilip** [bobby-brennan/rss-parser#46]
Hi ;

This library is _wonderful_, and works like a charm. But sometimes, the data returned by the server is a HTML page, when the same feed is actually viewable in a browser, for example this one:

http://www.jesusandmo.net/comic/feed/

Or this one :

https://www.blendernation.com/feed/

So I was wondering if there was some way to control the user agent to test if I can get the feed anyway... Maybe on the `request` side?

**Comment by bobby-brennan** [comment]
Currently there's no way to add headers to the HTTP request, but if you want to send a PR, I'd be happy to take a look. We set the user agent to 'rss-parser' right now.

Otherwise, I'd recommend doing the HTTP request yourself, and just using `parseString` instead of `parseURL`.

**Comment by yassinphilip** [comment]
This is what I use now to get the feed:


function getFeed (urlfeed, callback) {

```
  var options = {
    url: urlfeed,
    headers: {
      'User-Agent': 'Mozilla/5.0',
      'Content-Type': 'application/x-www-form-urlencoded'
    }
  };

  var req = request (options, urlfeed);
  var feedparser = new FeedParser ();
  var feedItems = [];
  req.on ('response', function (res) {
    var stream = this;
    if (res.statusCode === 200 && res.headers['content-type'].includes('xml')) {
      stream.pipe (feedparser);

    } else {
      callback (res.headers['content-type']);
      return;
    }
  });
  req.on ('error', function (res) {
    console.log ('getFeed: Error reading %s (%s) .', urlfeed, res);
  });
  feedparser.on ('readable', function () {
    try {
      var item = this.read ();
      if (item !== null) { //2/9/17 by DW
        feedItems.push (item);
      }
    }
    catch (err) {
      console.log ('getFeed: err.message == ' + err.message);
    }
  }).on ('end', function () {
    var meta = this.meta;
    callback ('Feed OK', feedItems, meta.title);
  }).on ('error', function (err) {
    // console.log ("getFeed: Error reading (%s) feed: %s.", urlfeed, err.message);
    callback ('Bad feed');
  });
}

router.get('/feed', function(req, res) {

  getFeed(req.query.feedurl, function (err, feedItems, feedTitle) {
        if (feedItems) {
            res.send({
                feedItems: feedItems,
                feedTitle: feedTitle
            });
        } else {
          res.send({error:err});
        }
    });

});
```

But http://www.blendernation.com/feed/ keeps sending back "text/html; charset=UTF-8" as `res.headers['content-type']`... Very puzzling ; when trying to wget the URI, it retrieves a proper XML/Atom file...? Any ideas as to exactly what either or both 'User-Agent' and 'Content-Type' values would permit the retrieval of the feed ?

**Comment by bobby-brennan** [comment]
I would suggest setting the [`Accept` header](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Accept) to tell it what content-type you want back. Probably `application/rss+xml` or `text/xml`

It probably makes sense for the parser to pass this header automatically as well.

**Comment by yassinphilip** [comment]
> It probably makes sense for the parser to pass this header automatically as well.

That would be great ; If I can help in any way (I'm thinking testing, to the extent of writing the proper tests) just ask.

**Comment by bobby-brennan** [comment]
I'm going to do this in v3, as changing the headers sent to the server might cause a different response.

Can you try it with the v3 branch? `npm install https://github.com/bobby-brennan/rss-parser#v3`

**Comment by bobby-brennan** [comment]
(note: the v3 API changed a bit, so [check the README](https://github.com/bobby-brennan/rss-parser/tree/v3))

**Comment by bobby-brennan** [comment]
This is merged in version 3.0.0

**Comment by bobby-brennan** [comment]
As pointed out in #111, adding `?format=xml` may fix this issue for some feeds.
